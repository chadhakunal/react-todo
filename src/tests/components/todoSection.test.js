import { shallow, mount } from "enzyme";
import TodoSection from "components/todoSection";
import TodoItem from "components/todoItem";

describe('TodoItem', () => {
  it('should render correctly with name, type and items', () => {
    const component = shallow(< TodoSection name="Hello" type="completed" items={[]} />);
    expect(component).toMatchSnapshot();
  });

  it('should call render all the TodoItems', ()=>{
    const component = shallow(< TodoSection name="Hello" type="completed" items={[
      {
        title: "Hello",
        description: "World"
      },
      {
        title: "Hello2",
        description: "World2"
      }
    ]} />);
    expect(component.find(TodoItem)).toHaveLength(2);
  })
})