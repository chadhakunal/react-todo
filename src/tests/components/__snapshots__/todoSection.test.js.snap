// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem should render correctly with name, type and items 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row mb-3 mt-3"
  >
    <big
      className="headers"
    >
      Hello
    </big>
  </div>
  <div
    className="container bordered-box pt-4"
    style={
      Object {
        "height": "60vh",
        "overflowY": "scroll",
      }
    }
  />
</div>
`;
